src/DeviceInterfaceValues.tsx

```diff
@@ -549,6 +552,33 @@ const UnsetDataModal = ({
     unsetInterfaceData(selectedEndpoint);
   };
 
+  useEffect(() => {
+    if (propertyData && interfaceDefinition) {
+      const mappingsWithUnsetAllowed = interfaceDefinition.mappings.filter(
+        (mapping) => mapping.allowUnset,
+      );
+
+      const matchesPattern = (endpoint: string, pattern: string): boolean => {
+        const regexPattern = pattern.replace(/%\{[^}]+\}/g, '[^/]+');
+        const regex = new RegExp(`^${regexPattern}$`);
+        return regex.test(endpoint);
+      };
```

Comment:
similarly here, `matchesPattern` is a bit misleading, This function doesn't just match a pattern, it first modifies it, then matches it from start to finish.
I suggest making it more explicit, like
```typescript
const matchMapping = (endpoint: string, mapping: AstarteMapping): boolean => {
  const endpointPattern = mapping.endpoint.replace(/%\{[^}]+\}/g, '[^/]+');
  const endpointRegex = new RegExp(`^${endpointPattern}$`);
  return endpointRegex.test(endpoint);
};

[...]

            mappingsWithUnsetAllowed.some((mapping) =>
              matchMapping(property.endpoint, mapping),
            )
```
Remember this is a pure function, so it's best declared outside of the React component (let's declare it just before `UnsetDataModalProps`)